checks_interface.md

```diff
@@ -0,0 +1,44 @@
+
+## Interface Explanation
+The PreCheck interface is implemented like so in the Types.go file. The Run function is the method used in order to actually execute the check hat implements this interface.
+
+```
+	type PreCheck interface {
+		Run(edges map[PromotionEdge]interface{}) error   
+	}
+```
+
+### How Checks Are Called
+A RunChecks method has been implemented which iterates over an input list of PreChecks and runs them on an input set of promotion edges. RunChecks then returns any errors that are returned from the PreChecks.
+
+```
+	func (sc *SyncContext) RunChecks(
+		edges map[PromotionEdge]interface{},
+		checks []PreCheck,
+	) error {
+		//Iterate over checks and run them on the edges
+	}
+```
+The *pull-k8s-cip* Prow job runs whenever a pull request attempts to modify the promoter manifests. It is only in this Prow job that the RunChecks function is called.
+
+### How To Add A Check
+In order to add a check, all you need to do is create a check type that implements the PreCheck interface.
+
+```
+	type foo struct {}
+	...
+	func (f *foo) Run(edges map[Promotionedge]interface{}) error
+```
+Then add that check type you've created to the input list of PreChecks for the RunChecks method, which is called in the cip.go file
+
+## Current Checks
+### ImageRemovalCheck
+Images that have been promoted are pushed to production; and once pushed to production, they should never be removed. The ImageRemovalCheck checks if any images are removed in the pull request by comparing the state of the promoter manifests in the pull request's branch to the master branch. Two sets of Promotion Edges are generated (one for both the master branch and pull request) and then compared to make sure that every destination image (defined by its image tag and digest) found in the master branch is found in the pull request.
+
+This method for detecting removed images should ensure that pull requests are only rejected if an image is completely removed from production, while still allowing edge cases. One example edge case is where a user has already promoted an image foo from registry A to registry B. Then in a later pull request, the user promotes the same image foo from registry C to registry B. Although image foo is removed from registry A, this pull request should be accepted because the same image is still being promoted, albeit from a new location. 
+
+### ImageSizeCheck
+The larger an image is, the more likely it is to be a security risk; and in general, it is bad practice to use unnecessarily large images. The ImageSizeCheck checks if any images to be promoted are larger than the defined maximum image size. The api calls that get image information from GCR also give information on the image size in bytes. These image sizes are recorded and then checked to make sure they are under the maximum size defined by the *--max-image-size* in MB. 
```

Comment:
I remember saying we should use MiB somewhere; I think MB is fine because that's what Docker also uses if you say `docker images`.